doc/kafka_rmq.md

```diff
@@ -0,0 +1,183 @@
+
+E.g.
+For publishing data using a producer which is defined for the stream router config with key `:default`, use send like this:
+
+`(send :default "test-topic" "key" "value")`
+
+`(send :default "test-topic" 1 "key" "value")`
+
+
+## Batch Consumption using Kafka Consumer API
+With Ziggurat version 3.5.1, both Kafka Streams API and Kafka Consumer API can be used to consume the messages in real
+time. Kafka Consumer API is an efficient way to consume messages from high throughput Kafka topics.
+
+With Kafka Streams API, one message is processed at a time. But, with Kafka Consumer API integration in Ziggurat,
+a user can consume messages in bulk and can control how many messages it wants to consume at a time. This batch size
+can be configured using max-poll-records config
+https://docs.confluent.io/current/installation/configuration/consumer-configs.html#max.poll.records.
+
+Like Streams, Ziggurat also provides the facility to specify multiple batch routes.
+
+### How to enable batch consumption in an actor?
+
+##### Changes required in config.edn
+```clojure
+:batch-routes {:restaurants-updates-to-non-personalized-es 
+                {:consumer-group-id          "restaurants-updates-consumer"
+                 :bootstrap-servers          "g-gojek-id-mainstream.golabs.io:6668"
+                 :origin-topic               "restaurant-updates-stream"}}
+```
+A full list of supported configs is given below. These configs can be added to `config.edn` as per the requirements.
+
+##### Call to Ziggurat Init Function
+```clojure
+(defn -main [& args]
+  (init/main {:start-fn      start
+              :stop-fn       stop
+              :stream-routes {:booking {:handler-fn (stream-deserializer/protobuf->hash
+                                                      stream-handler
+                                                      BookingLogMessage
+                                                      :booking)}}
+              :batch-routes  {:batch-consumer-1 {:handler-fn (batch-deserialzer/deserialize-batch-of-proto-messages
+                                                               batch-handler
+                                                               BookingLogKey
+                                                               BookingLogMessage
+                                                               :batch-consumer-1)}}
+              :actor-routes  [["v1/hello" {:get get-hello}]]}))
+```
+
+##### The Batch Handler
+```clojure
+(defn- single-message-handler
+  [message]
+  (log/info "Batch Message: " message))
+
+(defn batch-handler
+  [batch]
+  (log/infof "Received a batch of %d messages" (count batch))
+  (doseq [single-message batch]
+    (single-message-handler single-message))
+  (if (retry?)
+    (do (log/info "Retrying the batch..")
+        {:retry batch :skip []})
+    {:retry [] :skip []}))
+```
+
+##### List of all the supported configs for Batch Consumption
```

Comment:
why all other configs are part of this file which is named kafka_rmq.md, can you rename this to `channels.md` or rabbitmq_retry etc

Reply:
This file has information for both producer apis and batch consumption (including rabbitmq concepts ) hence named it kafka_rmq .